packages/core/src/utils.test.ts

```diff
@@ -308,13 +308,13 @@ describe('Core Utils', () => {
 
   test('Convert ArrayBuffer to hex string', () => {
     const input = new Uint32Array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
-    const output = arrayBufferToHex(input.buffer);
+    const output = arrayBufferToHex(input);
     expect(output).toBe('000102030405060708090a');
   });
 
   test('Convert ArrayBuffer to base-64 encoded string', () => {
     const input = new Uint32Array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
-    const output = arrayBufferToBase64(input.buffer);
+    const output = arrayBufferToBase64(input);
```

Comment:
Is this correct?

`new Uint8Array(input.buffer)` is not the same as `new Uint8Array(input)`
 
<img width="912" alt="Screenshot 2024-11-25 at 10 49 28 PM" src="https://github.com/user-attachments/assets/a38287dd-5878-4e2f-8a44-710581463811">

`new Uint8Array(input)` goes value by value and truncates them to fit into a byte, `new Uint8Array(input.buffer)` reinterprets the raw bytes as is, just applying a different "view" to it

Reply:
See #5599

Reply:
Yeah... I don't know what the correct behavior should be.

Changing from `input.buffer` to `input` is actually a revert back to the originally behavior.

Here is the finalized diff now, which just fixes the type error:

![image](https://github.com/user-attachments/assets/d962835e-474e-4414-91d7-f64fdefbffd9)

tbh, I don't know why TypeScript didn't catch the original errors a long time ago, but so it is

My intent with this PR was simply to upgrade dependencies without affecting the current behavior.  My preference would be to get this merged, and then have a conversation about what the "right" behavior should be.

#5599 looks reasonable to me.  I don't know if we actually need that in the wild though, or do we only need it for those 2 tests?  If it's just for the tests, maybe we should just simplify the behavior?

Reply:
I think this is the simple one because `TypedArray` extends `ArrayBuffer`, so the type will always accept both even though one is a view on top of the other. Either way, if you pass in an a `TypedArray` and not its underlying `.buffer`, you are truncating the output and it leads to an incorrect result.

The test expected an incorrect result for the `Uint32Array` we were passing in. In my opinion, this check for if the input is a view makes the implementation simpler. It's worth noting this is exported in core so I think we should make the function easy to use rather than dead simple but with a footgun as it currently is, given the test itself was even wrong.

Reply:
https://devblogs.microsoft.com/typescript/announcing-typescript-5-7/#support-for---target-es2024-and---lib-es2024

I think if we are planning to accept both `TypedArray` and `ArrayBuffer`, this should be `ArrayBufferLike` and not just `ArrayLike` to be more specific.

I think that covers both

Reply:
Actually, what happens if we have to use backwards incompatible types? That means users need to be on 5.7 too